tree/tree.go

```diff
@@ -16,28 +25,31 @@ type Tree interface {
 	DeleteVersion(version int64) error
 	GetImmutable() *ImmutableTree
 	GetImmutableAtHeight(version int64) (*ImmutableTree, error)
-	Version() int64
-	Hash() []byte
-	Iterate(fn func(key []byte, value []byte) bool) (stopped bool)
 }
 
-func NewMutableTree(db dbm.DB, cacheSize int) *MutableTree {
-	tree, err := iavl.NewMutableTree(db, cacheSize)
+func NewMutableTree(height uint64, db dbm.DB, cacheSize int, keepEvery, keepRecent int64) MTree {
+	tree, err := iavl.NewMutableTreeWithOpts(db, dbm.NewMemDB(), cacheSize, iavl.PruningOptions(keepEvery, keepRecent))
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = tree.LoadVersionForOverwriting(int64(height))
```

Comment:
LoadVersionOverwrite returns the version number of targetVersion.
Higher versions' data will be deleted.